ArjunaJTA/recovery/src/main/java/org/jboss/narayana/jta/quickstarts/recovery/JmsRecovery.java

```diff
@@ -270,86 +240,35 @@ private int consumeMessages(MessageConsumer consumer, long millis, int cnt) thro
     }
 
     public int consumeMessages(int cnt, long millis) throws Exception {
-        XAConnection connection = xacf.createXAConnection();
+        
         int msgCnt = 0;
 
-        try
-        {
+        try(XAConnection connection = xacf.createXAConnection();
+                XASession xaSession = connection.createXASession();
+                MessageConsumer xaConsumer = xaSession.createConsumer(queue);) {
             connection.start();
 
-            // create an XA JMS session and enlist the corresponding XA resource within a transaction
-            XASession xaSession = connection.createXASession();
+            // create an XA JMS session and enlist the corresponding XA resource within a
+            // transaction
```

Comment:
Clearer before (IMO)...but it's OK if you want to keep it like this :-)

Reply:
I agree, restored it. Thanks